### ISSUE TITLE
consider analysis feedback for enabled conflicting rules

### ISSUE THREAD

**Issue by pq** [dart-lang/linter#1896]
See https://github.com/dart-lang/linter/issues/1895 for an example (`unnecessary_finals` vs. `prefer_final_locals`).

/fyi @bwilkerson

**Comment by pq** [comment]
(Note to self: we might consider capturing, and at least documenting, these correspondences when we do "the big NNBD audit".)

Implementation bike-shed.

```dart

@incompatibleWith('unnecessary_finals')
class PreferFinalLocals extends LintRule implements NodeLintRule {
  ...
}
```

**Comment by bwilkerson** [comment]
I think that would be a good addition.

**Comment by pq** [comment]
Update: `LintRule`s now allow for specifying `incompatibleRules` which is enforced by DAS.

Next up:

- [ ] add validation to ensure `incompatibleRules` are properly defined in linter (e.g.,  describe actual rules and reference back to each other)
- [ ] audit rules and update incompatible ones

**Comment by pq** [comment]
Closing in favor of specific follow-on issues (#1964 and #1967).
